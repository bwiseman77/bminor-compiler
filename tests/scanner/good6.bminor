// strings
array auto boolean char else false for function if integer print return string true void while
// doube // comment
/* this doesnt fail but I think it should :)
for(integer x = 0; i < 10; i++) {
    print("hello world\n");
}
