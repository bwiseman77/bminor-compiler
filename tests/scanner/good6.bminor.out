-------------------------------------------
           token type  text
-------------------------------------------
 SINGLE_LINE_COMMENT  // strings
               ARRAY  array
                AUTO  auto
             BOOLEAN  boolean
                CHAR  char
                ELSE  else
               FALSE  false
                 FOR  for
            FUNCTION  function
                  IF  if
             INTEGER  integer
               PRINT  print
              RETURN  return
              STRING  string
                TRUE  true
                VOID  void
               WHILE  while
 SINGLE_LINE_COMMENT  // doube // comment
             DIVISON  /
      MULTIPLICATION  *
               IDENT  this
               IDENT  doesnt
               IDENT  fail
               IDENT  but
               IDENT  I
               IDENT  think
               IDENT  it
               IDENT  should
               COLON  :
       RIGHT_PARENTH  )
                 FOR  for
        LEFT_PARENTH  (
             INTEGER  integer
               IDENT  x
              ASSIGN  =
     INTEGER_LITERAL  0
           SEMICOLON  ;
               IDENT  i
                  LT  <
     INTEGER_LITERAL  10
           SEMICOLON  ;
               IDENT  i
           INCREMENT  ++
       RIGHT_PARENTH  )
        LEFT_BRACKET  {
               PRINT  print
        LEFT_PARENTH  (
      STRING_LITERAL  hello world

       RIGHT_PARENTH  )
           SEMICOLON  ;
       RIGHT_BRACKET  }
